flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -27,9 +27,30 @@ void TableColumn(TableCreationContext context) :
         UniqueKey(context.uniqueKeysList)
     |
         ComputedColumn(context)
+    |
+        Watermark(context)
     )
 }
 
+void Watermark(TableCreationContext context) :
+{
+    SqlIdentifier columnName;
+    SqlParserPos pos;
+    SqlNode watermarkStrategy;
+}
+{
+    <WATERMARK> <FOR>
+    columnName = CompoundIdentifier() {pos = getPos();}
+    <AS>
+    watermarkStrategy = Expression(ExprContext.ACCEPT_SUB_QUERY) {
+        if (context.watermark != null) {
+            throw new ParseException("Multiple WATERMARK statements is not supported yet.");
+        } else {
```

Comment:
The ParseException does not have position info, just to notes.

Reply:
Good point. I think we should follow `SqlUtil.newContextException` way.  The same to the insert partition exception.